src/main/perl/Lock.pm

```diff
@@ -246,63 +146,90 @@ sub unlock {
 
 =pod
 
-=item is_set
+=item is_set()
 
-Returns SUCCESS if lock is set by application instance, undef otherwise
+Returns B<SUCCESS> if lock is set by application instance, B<undef> otherwise.
 
 =cut
 
 sub is_set {
   my $self=shift;
   return SUCCESS if ($self->{LOCK_SET});
-  return undef;
+  return;
 }
 
+##############################################################################
+#
+# The following methods are provided for backwards compatibility only
```

Comment:
this is not what i consider `backwards compatibility`. these are noops that do something very very different from the original code.
i'd rather have you remove these then to keep them around implemented like this. if you want to keep them around, at least have them `->error('Danger danger: contact developers')` or something very obvious.

i grepped the quattor code, these 3 methods are only used in `CAF::Lock`, i vote to remove them.